Applications/Spire/Source/Ui/IconButton.cpp

```diff
@@ -46,14 +46,27 @@ IconButton::IconButton(QImage icon, Style style, QWidget* parent)
 }
 
 void IconButton::keyPressEvent(QKeyEvent* event) {
-  if(event->key() == Qt::Key_Enter || event->key() == Qt::Key_Return) {
+  if(!event->isAutoRepeat() && (event->key() == Qt::Key_Enter ||
+      event->key() == Qt::Key_Return)) {
```

Comment:
If you just want to assign the return/enter key to the button, I think calling setDefault with true is enough. I have tested it on the FlatButton, and it works.

Reply:
Is there side effects to this? The documentation makes it sound like there should only be one default button per window/dialog: [default property](https://doc.qt.io/qt-5/qpushbutton.html#default-prop)

Reply:
I am worried about this too. I have tested it using UiViewer, LoginUiTester and BookViewUiTester, and 
No bugs have been found so far. You can test it on your buttons.

Reply:
I would stick with overriding the input events because it makes the intent clearer. The QPushButton::setDefault(...) function applies some additional effects for parent dialogs and accessibility that look like they won't cause side effects (because we don't use the QDialog), but for clarity it's better handle the input logic explicitly.

Reply:
Yes, it's better to handle it explicitly to avoid potential side effects. I am wondering what the function isAutoRepeat() here do.

Reply:
If you press and hold a key down it generates multiple press events based on a repeat interval. This just checks to make sure that this event was sent as a single key press and wasn't a 'repeat' of a key that is being held down.